pkg/types/portmap.go

```diff
@@ -49,21 +49,38 @@ func (pps PortProtoSet) Equal(other PortProtoSet) bool {
 
 // NamedPortMultiMap may have multiple entries for a name if multiple PODs
 // define the same name with different values.
-type NamedPortMultiMap map[string]PortProtoSet
+type NamedPortMultiMap interface {
+	// GetNamedPort returns the port number for the named port, if any.
+	GetNamedPort(name string, proto uint8) (uint16, error)
+	// Len returns the number of Name->PortProtoSet mappings known.
+	Len() int
+	Equal(other NamedPortMultiMap) bool
+}
+
+func NewNamedPortMultiMap() namedPortMultiMap {
+	return make(namedPortMultiMap)
+}
+
+type namedPortMultiMap map[string]PortProtoSet
 
 // Equal returns true if the NamedPortMultiMaps are equal.
-func (npm NamedPortMultiMap) Equal(other NamedPortMultiMap) bool {
-	if len(npm) != len(other) {
+func (npm namedPortMultiMap) Equal(other NamedPortMultiMap) bool {
+	o, ok := other.(namedPortMultiMap)
```

Comment:
This strikes me as a bit odd. 

You take an interface in, but you pretty much force the interface to be a concrete type for comparison. 

Is this desired? If expose a way to obtain the {name, port} tuples in a generic way, then you can avoid the forcing the a type within the interface method implementation.

Reply:
Yeah, slightly odd but it looks like this is an intermediate state in the PR which is useful for transitioning to a subsequent implementation, and it's deleted by the end. There should be no risk of falsely asserting the wrong type :+1:

Reply:
Yup, reviewed this per commit, so this is even marked as outdated. Resolving